### PR TITLE
fix: stop leaking spatial_index and keep a stable schema on empty input

### PR DESCRIPTION
**Pull request by floriankrb**
The superob tabular filter added three helper columns (grid_index, spatial_index, distance) but only dropped two — spatial_index leaked into the
output as an undeclared column, breaking downstream variable enumeration (data carried one more column than the recorded variables).

This also makes the empty-input path schema-stable: instead of returning the frame untouched (skipping the aggregation reshape), an empty frame is
reordered to the same columns the non-empty path produces, so variable enumeration is consistent whether or not a date has observations.

- Drop spatial_index alongside grid_index and distance in forward().
- Add _empty_output_schema() and route the empty-input returns through it.
- Update the two superob tests (they had pinned the leaked column) and add a regression test asserting empty/non-empty schemas match and
spatial_index never leaks.

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/Contributor-License-Agreement.md)

### CODE REVIEW

File: src/anemoi/transform/filters/tabular/superob.py [hunk 1]
```diff
@@ -92,6 +125,6 @@ def forward(self, df: pd.DataFrame) -> pd.DataFrame:
         averaged_df = averaged_df[~averaged_df.index.duplicated(keep="first")]
         nearest_df = nearest_df[~nearest_df.index.duplicated(keep="first")]
         gridded_df = pd.concat([averaged_df, nearest_df], axis=1, join="inner").reset_index()
-        gridded_df = gridded_df.drop(columns=["grid_index", "distance"], errors="ignore")
+        gridded_df = gridded_df.drop(columns=["grid_index", "spatial_index", "distance"], errors="ignore")
```
Comment (aaron-hopkinson) [inline src/anemoi/transform/filters/tabular/superob.py:128]: We need to keep `spatial_index` – this is one of the main purposes of this filter (and is used by the `irregular_to_grid` filter)

File: src/anemoi/transform/filters/tabular/superob.py [hunk 2]
```diff
@@ -57,10 +57,43 @@ def __init__(
         self.columns_to_take_nearest = columns_to_take_nearest if columns_to_take_nearest else []
         self.columns_to_groupby = columns_to_groupby if columns_to_groupby else []
 
+    def _empty_output_schema(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Return an empty frame whose columns match the aggregated output.
+
+        The grid-assignment and aggregation path is skipped for empty inputs,
+        so this reorders the columns to mirror the schema produced for
+        non-empty inputs (grouped columns first, then the averaged columns,
+        then the nearest columns; the ``grid_index``, ``spatial_index`` and
```
Comment (aaron-hopkinson) [inline src/anemoi/transform/filters/tabular/superob.py:66]: See comment below – we _should_ include "spatial_index" here.


I'm not sure what we should do if `grid == "native"` though? (@pinnstorm?)
At the moment it's a no-op, so that would prevent the output from being stable...

File: tests/tabular_filters/test_superob.py [hunk 3]
```diff
@@ -58,6 +57,34 @@ def test_superob():
...
 
 
+def test_superob_empty_input_matches_non_empty_schema():
+    config = {
+        "grid": "o96",
+        "timeslot_length": 3600,
+        "columns_to_take_nearest": ["date"],
+        "columns_to_groupby": ["reportype"],
+    }
+    columns = ["date", "latitude", "longitude", "reportype", "obsvalue_rawbt_1"]
```
Comment (aaron-hopkinson) [inline tests/tabular_filters/test_superob.py:67]: (could just use the keys of the dict since they're ordered in python)

**Review by aaron-hopkinson — CHANGES_REQUESTED** [review]

File: src/anemoi/transform/filters/tabular/superob.py (hunk 2 above)
Comment (aaron-hopkinson) [reply src/anemoi/transform/filters/tabular/superob.py:66]: Maybe we need an `if` – if not native, add spatial_index to what you have here, otherwise don't

Comment (aaron-hopkinson) [reply src/anemoi/transform/filters/tabular/superob.py:66]: Alternatively, since native is just a noop, maybe we should just error in the constructor?

File: src/anemoi/transform/filters/tabular/superob.py [hunk 4]
```diff
@@ -57,10 +57,43 @@ def __init__(
...
+        -------
+        pandas.DataFrame
+            The empty frame with columns reordered to match the aggregated output schema.
+        """
+        grouped = set(self.columns_to_groupby)
+        nearest = set(self.columns_to_take_nearest)
+        ordered = (
+            list(self.columns_to_groupby)
+            + [c for c in df.columns if c not in (grouped | nearest)]
+            + list(self.columns_to_take_nearest)
```
Comment (floriankrb) [inline src/anemoi/transform/filters/tabular/superob.py:86]: todo: add spatial_index here.?

File: src/anemoi/transform/filters/tabular/superob.py [hunk 5]
```diff
@@ -57,10 +57,43 @@ def __init__(
...
+        )
+        return df.reindex(columns=ordered)
+
     def forward(self, df: pd.DataFrame) -> pd.DataFrame:
-        if self.grid == "native" or len(df) == 0:
+        if self.grid == "native":
             return df
```
Comment (floriankrb) [inline src/anemoi/transform/filters/tabular/superob.py:92]: remove this and add raise ValueError in init: grid=native not supported by supperob filter.

File: src/anemoi/transform/filters/tabular/superob.py (hunk 1 above)
Comment (floriankrb) [reply src/anemoi/transform/filters/tabular/superob.py:128]: keep it.

Comment (floriankrb) [reply src/anemoi/transform/filters/tabular/superob.py:128]: + add option to drop/keep the spatial_index : 'add_spatial_index' (default = False)
